### PR TITLE
COVERAGE Allow component management

### PR DESCRIPTION
**Pull request by neolynx**
Fixes #

## Requirements

All new code should be covered with tests, documentation should be updated. CI should pass.

## Checklist

- [ ] unit-test added (if change is algorithm)
- [ ] functional test added/updated (if change is functional)
- [ ] man page updated (if applicable)
- [ ] bash completion updated (if applicable)
- [ ] documentation updated
- [ ] author name in `AUTHORS`

### CODE REVIEW

File: api/publish.go [hunk 1]
```diff
@@ -403,3 +493,368 @@ func apiPublishDrop(c *gin.Context) {
 		return &task.ProcessReturnValue{Code: http.StatusOK, Value: gin.H{}}, nil
 	})
 }
+
+// @Router /api/publish/{prefix}/{distribution}/sources [post]
```
Comment (neolynx) [inline api/publish.go:497]: swagger doc

File: api/publish.go [hunk 2]
```diff
@@ -403,3 +493,368 @@ func apiPublishDrop(c *gin.Context) {
...
+		err = collection.Update(published)
+		if err != nil {
+			return &task.ProcessReturnValue{Code: http.StatusInternalServerError, Value: nil}, fmt.Errorf("unable to save to DB: %s", err)
+		}
+
+		return &task.ProcessReturnValue{Code: http.StatusOK, Value: published}, nil
+	})
+}
+
+// @Router /api/publish/{prefix}/{distribution}/sources [delete]
```
Comment (neolynx) [inline api/publish.go:635]: swagger doc

File: api/publish.go [hunk 3]
```diff
@@ -403,3 +493,368 @@ func apiPublishDrop(c *gin.Context) {
...
+		err = collection.Update(published)
+		if err != nil {
+			return &task.ProcessReturnValue{Code: http.StatusInternalServerError, Value: nil}, fmt.Errorf("unable to save to DB: %s", err)
+		}
+
+		return &task.ProcessReturnValue{Code: http.StatusOK, Value: published}, nil
+	})
+}
+
+// @Router /api/publish/{prefix}/{distribution}/sources/{component} [put]
```
Comment (neolynx) [inline api/publish.go:670]: swagger doc

File: api/publish.go [hunk 4]
```diff
@@ -403,3 +493,368 @@ func apiPublishDrop(c *gin.Context) {
...
+		err = collection.Update(published)
+		if err != nil {
+			return &task.ProcessReturnValue{Code: http.StatusInternalServerError, Value: nil}, fmt.Errorf("unable to save to DB: %s", err)
+		}
+
+		return &task.ProcessReturnValue{Code: http.StatusOK, Value: published}, nil
+	})
+}
+
+// @Router /api/publish/{prefix}/{distribution}/sources/{component} [delete]
```
Comment (neolynx) [inline api/publish.go:733]: swagger doc

File: api/publish.go [hunk 5]
```diff
@@ -403,3 +493,368 @@ func apiPublishDrop(c *gin.Context) {
...
+		err = collection.Update(published)
+		if err != nil {
+			return &task.ProcessReturnValue{Code: http.StatusInternalServerError, Value: nil}, fmt.Errorf("unable to save to DB: %s", err)
+		}
+
+		return &task.ProcessReturnValue{Code: http.StatusOK, Value: published}, nil
+	})
+}
+
+// @Router /api/publish/{prefix}/{distribution}/update [post]
```
Comment (neolynx) [inline api/publish.go:774]: swagger doc

File: api/publish.go [hunk 6]
```diff
@@ -403,3 +493,368 @@ func apiPublishDrop(c *gin.Context) {
...
+		err = collection.Update(published)
+		if err != nil {
+			return &task.ProcessReturnValue{Code: http.StatusInternalServerError, Value: nil}, fmt.Errorf("unable to save to DB: %s", err)
+		}
+
+		return &task.ProcessReturnValue{Code: http.StatusCreated, Value: published}, nil
+	})
+}
+
+// @Router /api/publish/{prefix}/{distribution}/sources [get]
```
Comment (neolynx) [inline api/publish.go:553]: swagger doc

Comment (cfiehe) [reply api/publish.go:594]: I have started to add the missing Swagger doc yesterday.

**Pull request closed without merging**